### PR TITLE
Feature/search box with speech capability

### PR DESCRIPTION
**Pull request by imsauravsingh**
Component Details:
Molecules: Search component 
Issue: https://github.com/pagesource/atomic-react-components/issues/100

### CODE REVIEW

File: lib/components/molecules/Search/Search.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    if (e.key !== 'undefined' && e.key !== 'Enter') return false;
+    onSearchClick();
+  };
+
+  const searchList = () => {
+    return (
+      <List ListType="ul">
+        {searchLists.map((val, index) => {
+          return (
+            <li className="search-list" key={index}>
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js]: Using index in key is bad practice and linters also catch it.

File: lib/components/molecules/Search/Search.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    if (inputValue.trim()) {
+      const lists = dataLists.filter(val => {
+        return val.toLowerCase().indexOf(inputValue.toLowerCase()) !== -1;
+      });
+      setSearchList(lists);
+    }
+    return false;
+  };
+
+  const keyPress = e => {
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js]: You can do key press on list items too and on search field too, which one it is?

File: lib/components/molecules/Search/Search.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
+// @flow
+/**
+ *
+ * Search
+ *
+ */
+import React, { useEffect, useState, useCallback } from 'react';
+import styled from 'styled-components';
+import classnames from 'classnames';
+import Input from '../../atoms/Input';
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js:11]: There should be a space/line between external imports and your own modules.

File: lib/components/molecules/Search/Search.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      </List>
+    );
+  };
+
+  return (
+    <div className={classnames(styles.default, className)}>
+      <div className="search-box">
+        <Input
+          aria-label="Search"
+          {...inputProps}
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js:88]: {...inputProps} Shouldn't this be set last?

File: lib/components/molecules/Search/Search.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+          className={classnames('search-txt', inputProps.className)}
+          value={inputValue}
+          onChange={onInputChange}
+          onKeyPress={keyPress}
+        />
+        <div className="mic-box">
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js:95]: This should have been an icon button.

File: lib/components/molecules/Search/Search.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+          <div className="mic-box-2">
+            <span className="search-mic" />
+          </div>
+        </div>
+        <Button onClick={onSearchClick} className="search-btn-container" type="button">
+          <div className="search-btn-box">
+            <span className="search-btn" />
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js:102]: We have an Icon component? Use of Div inside Button is wrong

File: lib/components/molecules/Search/Search.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+          value={inputValue}
+          onChange={onInputChange}
+          onKeyPress={keyPress}
+        />
+        <div className="mic-box">
+          <div className="mic-box-2">
+            <span className="search-mic" />
+          </div>
+        </div>
+        <Button onClick={onSearchClick} className="search-btn-container" type="button">
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js]: search-btn-container feels like a poorly named class

File: lib/components/molecules/Search/Search.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+          <div className="search-btn-box">
+            <span className="search-btn" />
+          </div>
+        </Button>
+      </div>
+      {searchList()}
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js]: Why not create this a functional component? You will be able to use it declarative way.

File: lib/components/molecules/Search/Search.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    </div>
+  );
+};
+
+Search.defaultProps = {
+  className: '',
+  inputProps: {},
+  searchLists: ['test1', 'test2'],
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.js]: No dummy data in default props

File: lib/components/molecules/Search/Search.story.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+// Import Styled Component to showcase variations
+import Search, { SearchVanilla } from '.';
+const ListItem = ({ children }) => children;
+
+// TODO: indentation is not working fine in editor. Need to check for .hbs files
+// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
+storiesOf('Molecules', module)
+  .addParameters({ jest: ['Search', 'SearchVanilla'] })
+  .add('CTA Knobs', () => (
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.story.js]: Change this to relevant name

File: lib/components/molecules/Search/Search.story.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import { storiesOf } from '@storybook/react';
+// import { action } from '@storybook/addon-actions';
+import defaultConfig from './Search.mock';
+
+// Import Styled Component to showcase variations
+import Search, { SearchVanilla } from '.';
+const ListItem = ({ children }) => children;
+
+// TODO: indentation is not working fine in editor. Need to check for .hbs files
+// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.story.js]: Remove // $FlowFixMe line

File: lib/components/molecules/Search/Search.story.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+storiesOf('Molecules', module)
+  .addParameters({ jest: ['Search', 'SearchVanilla'] })
+  .add('CTA Knobs', () => (
+    <SearchVanilla
+      {...defaultConfig}
+      className={`hide-default-sample ${defaultConfig.className}`}
+    />
+  ))
+  .add('CTA', () => (
+    <Search {...defaultConfig} renderListItem={item => <ListItem key={item}>{item}</ListItem>} />
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.story.js:20]: Nice

File: lib/components/molecules/Search/Search.style.js [hunk 13]
```diff
@@ -0,0 +1,91 @@
+import { css } from 'styled-components';
+
+export default css`
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.style.js:3]: This is so poorly written CSS doesn't make use of even a single theme value

File: lib/components/molecules/Search/Search.style.js [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+  .search-box,
+  .search-list {
+    background: #fff;
+    display: flex;
+    border-radius: 8px;
+    border: 1px solid #dfe1e5;
+    box-shadow: none;
+    height: 39px;
+    width: 638px;
+    border-radius: 24px;
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.style.js:14]: How come same selector has border-radius 2 times

File: lib/components/molecules/Search/Search.style.js [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+    z-index: 3;
+    height: 44px;
+    margin: 0 auto;
+  }
+  .search-list {
+    border-radius: 5px;
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.style.js:20]: Why does .search-list has border-radius?

File: lib/components/molecules/Search/Search.style.js [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+    margin-top: 10px;
+    line-height: 40px;
+    padding-left: 12px;
+  }
+  .mic-box {
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.style.js:25]: Wrong use of flex

File: lib/components/molecules/Search/Search.style.js [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+    float: left;
+    padding-left: 10px;
+    color: black;
+    font-size: 16px;
+    transition: 0.4s;
+    line-height: 40px;
+    width: 86%;
+  }
+  .search-btn-container {
+    flex: 0 0 auto;
```
Comment (vinodloha) [inline lib/components/molecules/Search/Search.style.js:63]: Too many unnecessary props

File: lib/components/molecules/Search/tests/Search.test.js [hunk 18]
```diff
@@ -0,0 +1,14 @@
+import { shallow } from 'enzyme';
+
+import Search from '../index';
+
+describe('<Search />', () => {
```
Comment (vinodloha) [inline lib/components/molecules/Search/tests/Search.test.js:5]: You had to do TDD

File: lib/components/molecules/Search/Search.js (hunk 2 above)
Comment (imsauravsingh) [reply lib/components/molecules/Search/Search.js]: It's in search field.

File: lib/components/molecules/Search/Search.js (hunk 6 above)
Comment (imsauravsingh) [reply lib/components/molecules/Search/Search.js:102]: we don't have Icon component.

**Pull request closed without merging**